File: GameOfLife/Engine/Cell.cs
using System.Collections.Generic;
using System.Drawing;

namespace GameOfLife.Engine {
  public class Cell {
    #region Private Variables
    private int x;
    private int y;    
    #endregion

    #region Constructor

    public Cell(int xPos, int yPos, int aWidth, int aHeight, int offset) {
      Width = aWidth;
      Height = aHeight;
      Neighbours = new List<Cell>();
      x = xPos;
      y = yPos;
      CellGraphic = new Rectangle(x * Width, y * Height, Width, Height);
    }

    #endregion

    #region Properties

    public int Width { get; private set; }
    public int Height { get; private set; }
    public bool Alive { get; set; }
    public List<Cell> Neighbours { get; set; }
    public Rectangle CellGraphic { get; set; }

    #endregion

    #region Public Methods

    public Cell NextGeneration() {
      var nextGeneration = new Cell(x, y, Width, Height, 10) { Neighbours = Neighbours };
      int liveNeighbours = 0;
      foreach (Cell neighbour in Neighbours) {
        if (neighbour.Alive) {
          liveNeighbours++;
        }
      }
      // Check if the nextGeneration is alive
      if (Alive) {
        nextGeneration.Alive = liveNeighbours == 2 || liveNeighbours == 3;        
      }
      else {
        // Check if the nextGeneration is reborn
        nextGeneration.Alive = liveNeighbours == 3;
      }
      return nextGeneration;
    }

    #endregion
  }
}

File: GameOfLife.Gtk/Cell.cpp
#include "Cell.h"

Cell::Cell(int aX, int aY, int aWidth, int aHeight)
{
  x = aX;
  y = aY;
  width = aWidth;
  height = aHeight;
  alive = true;
}

Cell::~Cell(void)
{
}

bool Cell::IsAlive()
{
  return alive;
}

Cell Cell::NextGeneration()
{	
  Cell nextGen = Cell(x, y, width, height);
  nextGen.SetNeighbours(neighbours);
  int liveNeighbours = 0;
  for (std::list<Cell*>::iterator i = neighbours.begin(); i != neighbours.end(); i++)
  {
    if ((*i)->IsAlive())
    {
      liveNeighbours++;
    }
  }
  // Check if the nextGeneration is alive
  if (alive) {
    nextGen.SetAlive(liveNeighbours == 2 || liveNeighbours == 3);
  }
  else {
    // Check if the nextGeneration is reborn
    nextGen.SetAlive(liveNeighbours == 3);
  }
  return nextGen;
}

void Cell::SetNeighbours(std::list<Cell*> newNeighbours)
{
  neighbours = newNeighbours;
}

void Cell::SetAlive(bool isAlive)
{
  alive = isAlive;
}

void Cell::ClearNeighbours()
{
  neighbours.clear();
}

std::list<Cell*> Cell::GetNeighbours()
{
  return neighbours;
}

int Cell::GetX()
{
  return x;
}

int Cell::GetY()
{
  return y;
}

int Cell::GetWidth()
{
  return width;
}

int Cell::GetHeight()
{
  return height;
}

void Cell::AddNeighbour(Cell* neighbour)
{
  neighbours.push_back(neighbour);
}